libs/feature-recipe-search/src/lib/feature-recipe-search/feature-recipe-search.component.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ElementRef,
+  HostListener,
+  ViewChild,
+  inject,
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { FormControl, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatInputModule } from '@angular/material/input';
+import { RecipesUiActions } from '@hoa-recipes/data-access-recipes';
+import { Store } from '@ngrx/store';
+
+@Component({
+  selector: 'hoa-recipes-feature-recipe-search',
+  standalone: true,
+  imports: [
+    MatFormFieldModule,
+    FormsModule,
+    MatInputModule,
+    ReactiveFormsModule,
+  ],
+  templateUrl: './feature-recipe-search.component.html',
+  styleUrl: './feature-recipe-search.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FeatureRecipeSearchComponent {
+  private store = inject(Store);
+
+  @ViewChild('searchInput') searchInput!: ElementRef<HTMLInputElement>;
+
+  formControl = new FormControl('');
+
+  @HostListener('window:keydown', ['$event'])
+  onKeyDownAltM(e: KeyboardEvent) {
+    if (e.ctrlKey && e.key === 'k') {
+      e.preventDefault();
+      this.searchInput.nativeElement.focus();
+    }
+  }
+
+  constructor() {
+    this.formControl.valueChanges
+      .pipe(takeUntilDestroyed())
```

Comment:
adding a `debounceTime` for this purpose is a good practice

Reply:
@DDonochVA fixed, merging now